pallets/parachain-staking/src/lib.rs

```diff
@@ -1532,6 +1527,10 @@ pub mod pallet {
 					// clean up storage items that we no longer need
 					<DelayedPayouts<T>>::remove(paid_for_round);
 					<Points<T>>::remove(paid_for_round);
+
+					// remove up to 1000 candidates that did not produce any blocks for
+					// the given round
+					let _ = <AtStake<T>>::clear_prefix(paid_for_round, 1000, None);
```

Comment:
We could use the value from `TopSelected` but maybe we don't want an extra read. Alternatively, a constant for this would be good.

If we make our rounds longer, this problem should mostly go away (except for very poorly performing collators...)